v1alpha1/meshmodel.proto

```diff
@@ -619,3 +619,35 @@ message TrafficTarget {
 	// Rules are the traffic rules to allow (HTTPRoutes).
 	repeated TrafficTargetRule rules = 4;
 }
+
+message ServiceCanary {
+	// Name is the name of service canary.
+	string name = 1;
+
+	// Priority must be [1, 9], the default is 5 if user does not set it.
+	// The smaller number get higher priority.
+	// The order is sorted by name alphabetically in the same priority.
+	int32 priority = 2;
+
+	// Selector is the service selector to choose
+	// service instances.
+	ServiceSelector selector = 3;
```

Comment:
Is ServiceSelector should be a pair of the service and an array of the service instance? why don't you repeat `ServiceSelctor`?